nexus/src/external_api/params.rs

```diff
@@ -25,6 +25,41 @@ pub struct SiloCreate {
     pub discoverable: bool,
 }
 
+// Silo identity providers
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct SiloSamlIdentityProviderCreate {
+    #[serde(flatten)]
+    pub identity: IdentityMetadataCreateParams,
+
+    pub silo_id: Uuid,
+
+    /// url where identity provider metadata descriptor is
+    pub idp_metadata_url: String,
+
+    /// idp's entity id
+    pub idp_entity_id: String,
+
+    /// sp's client id
+    pub sp_client_id: String,
+
+    /// service provider endpoint where the response will be sent
+    pub acs_url: String,
+
+    /// service provider endpoint where the idp should send log out requests
+    pub slo_url: String,
+
+    /// customer's technical contact for saml configuration
+    pub technical_contact_email: String,
+
+    /// optional request signing public certificate (base64 encoded der file)
+    pub public_cert: Option<String>,
+
+    /// optional request signing private key (base64 encoded der file)
+    #[serde(skip_serializing)]
```

Comment:
It's meant to enforce that clients can POST in a key, but not see it on a GET.

Reply:
I'm curious about what you mean by "the client never sends it" - what do you mean by this?

Reply:
We have `#[serde(skip_serializing)]` this field -- doesn't that mean if the client sends this type to the server, that field will be missing?

Reply:
> It's meant to enforce that clients can POST in a key, but not see it on a GET.

I see -- I don't think that's quite right, because a client (as in the test suite) could totally use the `params` struct directly to send the type to the server, and that would do the wrong thing.

This sounds like a good use case for having a separate `view` type that describes whatever we return (even if it's mostly the same fields).

Reply:
added view type in 5f352d05, plus removed the skip_serializing bit.